prqlc/prql-compiler/src/semantic/std.prql

```diff
@@ -215,6 +215,15 @@ let from_text = input<text> `noresolve.format`:csv -> <relation> internal from_t
 ## String functions
 let lower = column -> <text> internal std.lower
 let upper = column -> <text> internal std.upper
+let ltrim = column -> <text> internal std.ltrim
+let rtrim = column -> <text> internal std.rtrim
+let trim = column -> <text> internal std.trim
+let length = column -> <int> internal std.length
```

Comment:
Fine for now, but eventually for `length` probably we want it within a `string` namespace; given how general the term is

Reply:
Yes I'm planning to first open a PR to add the `math` module
And for this one I plan to add a `str` module
Would that be ok?

So we would have `derive { x | str.lower | str.endswith "foo" }`

Reply:
A module `str` has been added. Do you prefer `string`?

Reply:
Perfect re a `str` module!

No particularly strong view on `str` vs `string`. Possibly we should pick the one that is _not_ the name of the type.

I'd also previously thought that "string" was a bit too "engineer-y", and most folks think a string is something used for tying. So we currently have `from_text`, for example. But I'm not sure we're going to win that battle! So I'm fine with any of those.

Reply:
It seems like most languages prefer `string` or `String` over `str` and `String` or `text`.

- C# uses `string.Format`
- D uses `std.string.isNumeric("123")`
- Elixir uses `String.first("elixir")`
- Go uses `strings.ToLower("Hello")`
- Java uses `String.format`
- JavaScript uses `String.fromCharCode(65)`
- Kotlin uses `String.format("Hello")`
- Python uses `str.format`
- Ruby uses `String::try_convert('Hello')`
- Scala uses `String.format("Hello")`